common/retry/retry.go

```diff
@@ -76,17 +76,36 @@ func (cfg Config) Run(ctx context.Context, fn func(context.Context) error) error
 	}
 }
 
+// Backoff is a self contained backoff time calculator. This struct should be
+// passed around as a copy as it changes its own fields upon any Backoff call.
+// Backoff is not thread safe. For now only a Linear backoff call is
+// implemented and the Exponential call will be implemented when needed.
 type Backoff struct {
+	// Initial time to wait. A Backoff call will change this value.
 	InitialBackoff time.Duration
-	MaxBackoff     time.Duration
-	Multiplier     float64
+	// Maximum time returned.
+	MaxBackoff time.Duration
+	// For a Linear backoff, InitialBackoff will be multiplied by Multiplier
+	// after each call.
+	Multiplier float64
 }
 
+// Linear Backoff returns a linearly increasing Duration.
+//  n = n * Multiplier.
+// the first value of n is InitialBackoff. n is maxed by MaxBackoff.
 func (lb *Backoff) Linear() time.Duration {
+	if lb.InitialBackoff > lb.MaxBackoff {
+		panic("InitialBackoff > MaxBackoff, did you forgot setting the seconds ?")
```

Comment:
I added this panic to avoid future mistakes, since the struct is only manually set, this panic should only happen during dev time

Reply:
Actually this panic is not very handy, let's count on reviews 🙂